README.md

```diff
@@ -324,6 +332,393 @@ The spreadsheet data will always be deleted automatically after 72 hours if it i
 $geocoder->deleteList(11950669);
 ```
 
+### Distance calculations
+
+Calculate distances from a single origin to multiple destinations, or compute full distance matrices.
+
+#### Coordinate format with custom IDs
+
+You can add custom identifiers to coordinates using the `lat,lng,id` format. The ID will be returned in the response, making it easy to match results back to your data:
+
+```php
+// String format with ID
+'37.7749,-122.4194,warehouse_1'
+
+// Array format with ID
+[37.7749, -122.4194, 'warehouse_1']
+
+// The ID is returned in the response:
+// [
+//     'query' => '37.7749,-122.4194,warehouse_1',
+//     'location' => [37.7749, -122.4194],
+//     'id' => 'warehouse_1',
+//     'distance_miles' => 3.2,
+//     'distance_km' => 5.1
+// ]
+```
+
+#### Distance mode and units
+
+The SDK provides enums for type-safe distance configuration:
+
+```php
+use Geocodio\Enums\DistanceMode;
+use Geocodio\Enums\DistanceUnits;
+use Geocodio\Enums\DistanceOrderBy;
+use Geocodio\Enums\DistanceSortOrder;
+
+// Available modes
+DistanceMode::Straightline  // Default - great-circle (as the crow flies)
+DistanceMode::Driving       // Road network routing with duration
+DistanceMode::Haversine     // Alias for Straightline (backward compat)
```

Comment:
To avoid confusion, let's remove Haversine from the README (the constant can still exist in the codebase but with PHPDoc explaining that it's an alias for Straightline)